### PR TITLE
Changes to OAuth2TokenCache internals and api surface to support MSAL getAccounts()

### PR DESCRIPTION
**Pull request by iambmelt**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java [hunk 1]
```diff
@@ -161,6 +162,182 @@ public void saveTokens(
...
+        // Get all of the Accounts for this environment
+        final List<Account> accountsForEnvironment =
+                mAccountCredentialCache.getAccountsFilteredBy(
+                        null, // wildcard (*) homeAccountId
+                        environment,
+                        null // wildcard (*) realm
+                );
+
+        // Declare a List to hold the MicrosoftStsAccounts
+        final List<Account> microsoftStsAccounts = new ArrayList<>();
```
Comment (shoatman) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: We probably shouldn't assume these accounts are MSTS accounts.  What about ADFS, B2C?

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java [hunk 2]
```diff
@@ -161,6 +162,182 @@ public void saveTokens(
...
+
+        // Grab all of the Accounts supplied by either the v1 or v2 endpoint
+        for (final Account account : accountsForEnvironment) {
+            if (MicrosoftAccount.AUTHORITY_TYPE_V1_V2.equals(account.getAuthorityType())) {
+                microsoftStsAccounts.add(account);
+            }
+        }
+
+        // Grab the Credentials for this app...
+        final List<Credential> appCredentials =
```
Comment (shoatman) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: appCredentials is a potentially misleading name.... do you mean get all credentials currently in the cache?  If so... perhaps... allCredentials?

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java [hunk 3]
```diff
@@ -161,6 +162,182 @@ public void saveTokens(
...
+                        environment,
+                        CredentialType.RefreshToken,
+                        clientId,
+                        null, // realm
+                        null // target
+                );
+
+        // For each Account with an associated RT, add it to the result List...
+        for (final Account account : microsoftStsAccounts) {
+            if (accountHasToken(account, appCredentials)) {
```
Comment (shoatman) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: rather than accountHasToken....accountHasCredential

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: Resolved via c8cffe58b5905762221cd6ccce2d5a95db888404

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java (hunk 1 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: Resolved via ea7e9ee4b7eb23135ad0223021b7dcfa5175e817

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java (hunk 2 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: Going to leave as-is; these credentials are loaded only for the current clientId (see param to `getCredentialsFilteredBy`)

**Pull request merged**